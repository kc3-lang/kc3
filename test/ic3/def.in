quote def zero = 0
def zero = 0
quote zero
zero
quote def one_two_three = [1, 2, 3]
def one_two_three = [1, 2, 3]
quote one_two_three
one_two_three
quote def double = fn (x) { x * 2 }
def double = fn (x) { x * 2 }
quote double
double
quote double(200)
double(200)
quote double(zero)
double(zero)
quote def double_tuple = macro (x) do
  quote do
    x = ^ unquote(x)
    {x, x}
  end
end
def double_tuple = macro (x) do
  quote do
    x = ^ unquote(x)
    {x, x}
  end
end
quote double_tuple
double_tuple
quote double_tuple(200)
double_tuple(200)
quote double_tuple(zero)
double_tuple(zero)
quote double_tuple(one_two_three)
double_tuple(one_two_three)
quote def reverse = fn {
  (x) { reverse(x, []) }
  ([], acc) { acc }
  ([a | b], acc) { reverse(b, [a | acc]) }
}
def reverse = fn {
  (x) { reverse(x, []) }
  ([], acc) { acc }
  ([a | b], acc) { reverse(b, [a | acc]) }
}
quote reverse
reverse
quote reverse([1, 2, 3])
reverse([1, 2, 3])
quote def reverse = fn {
  (x) { reverse(x, []) }
  ([], acc) { [:reversed | acc] }
  ([a | b], acc) { reverse(b, [a | acc]) }
}
def reverse = fn {
  (x) { reverse(x, []) }
  ([], acc) { [:reversed | acc] }
  ([a | b], acc) { reverse(b, [a | acc]) }
}
quote reverse
reverse
quote reverse([1, 2, 3])
reverse([1, 2, 3])
quote def reverse = fn {
  (x) { reverse(x, []) }
  ([], acc) { acc }
  ([a | b], acc) { reverse(b, [a | acc]) }
}
def reverse = fn {
  (x) { reverse(x, []) }
  ([], acc) { acc }
  ([a | b], acc) { reverse(b, [a | acc]) }
}
quote reverse
reverse
quote reverse([1, 2, 3])
reverse([1, 2, 3])
quote def reverse = fn (x) { List.reverse(x) }
def reverse = fn (x) { List.reverse(x) }
quote reverse
reverse
quote reverse([1, 2, 3])
reverse([1, 2, 3])
quote module()
module()
quote search_modules()
search_modules()
quote reverse
reverse
quote def reverse = fn (x) { [:reversed | List.reverse(x)] }
def reverse = fn (x) { [:reversed | List.reverse(x)] }
quote reverse
reverse
quote reverse([1, 2, 3])
reverse([1, 2, 3])
quote module()
module()
quote search_modules()
search_modules()
quote reverse
reverse
quote def reverse = fn (x) { List.reverse(x) }
def reverse = fn (x) { List.reverse(x) }
quote reverse
reverse
quote reverse([1, 2, 3])
reverse([1, 2, 3])
quote module()
module()
quote search_modules()
search_modules()
quote reverse
reverse
