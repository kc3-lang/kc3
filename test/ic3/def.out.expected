def zero = 0
C3.zero
zero
0
def one_two_three = [1, 2, 3]
C3.one_two_three
one_two_three
[1, 2, 3]
def double = fn (x) { x * 2 }
C3.double
double
fn (x) { x * 2 }
double(200)
400
double(zero)
0
def double_tuple = macro (x) do
  {x, x}
end
C3.double_tuple
double_tuple
macro (x) do
  {x, x}
end
double_tuple(200)
{200, 200}
double_tuple(zero)
{0, 0}
double_tuple(one_two_three)
{[1, 2, 3], [1, 2, 3]}
def reverse = fn (x) { List.reverse(x) }
C3.reverse
C3.reverse
fn (x) { List.reverse(x) }
reverse([1, 2, 3])
[3, 2, 1]
def reverse = fn (x) { [:reversed | List.reverse(x)] }
C3.reverse
C3.reverse
fn (x) { [:reversed | List.reverse(x)] }
reverse([1, 2, 3])
[:reversed, 3, 2, 1]
def reverse = fn (x) { List.reverse(x) }
C3.reverse
C3.reverse
fn (x) { List.reverse(x) }
reverse([1, 2, 3])
[3, 2, 1]
